LibreNMS/Alert/Transport/Xmatters.php

```diff
@@ -0,0 +1,121 @@
+<?php
+/**
+ * Xmatters API Transport
+ *
+ * @author Sigurdur Stefan <https://github.com/LoveSkylark>
+ * @copyright 2023 Sigurdur Stefan
+ * @license GPL
+ */
+
+namespace LibreNMS\Alert\Transport;
+
+use LibreNMS\Alert\Transport;
+use LibreNMS\Exceptions\AlertTransportDeliveryException;
+use LibreNMS\Util\Http;
+
+class Xmatters extends Transport
+{
+    public function deliverAlert(array $alert_data): bool
+    {
+        $url = $this->config['xmatters_url'] . '/api/integration/1/functions/' . $this->config['xmatters_function'] . '/triggers';
+
+        $device_url = \Config::get('base_url');
+        $device_groups = \DeviceCache::get($alert_data['device_id'])->groups->pluck('name');
+
+        // Build request data
+        $data = [
+            'properties' => $alert_data,
+            'recipients' => $alert_data['contacts'],
+            'priority' => $this->mapSeverityToPriority($alert_data['severity']),
+        ];
+
+        $data['properties']['event_state'] = $this->mapStateToEventState($alert_data['state']);
+        $data['properties']['instance_url'] = $device_url;
+        $data['properties']['device_url'] = ($device_url . '/device/' . $alert_data['device_id']);
+        $data['properties']['alert_url'] = ($device_url . '/device/' . $alert_data['device_id'] . '/alerts/' . $alert_data['alert_id']);
+        $data['properties']['hostname'] = $alert_data['sysName'];
+        $data['properties']['ip'] = $alert_data['hostname'];
+        $data['properties']['device_groups'] = $device_groups;
+
+        $res = Http::client()
+            ->withBasicAuth($this->config['xmatters_api_key'], $this->config['xmatters_api_secret'])
+            ->acceptJson()
+            ->post($url, $data);
+
+        if ($res->successful()) {
+            return true;
+        }
+
+        throw new AlertTransportDeliveryException($alert_data, $res->status(), $res->body(), $alert_data['msg'], $data);
+    }
+
+    protected function mapSeverityToPriority(string $severity): string
```

Comment:
I think you could probably use Enums here. See an example on: https://github.com/librenms/librenms/blob/298d217a8b4403196ee1d62d1a2c41637ad66342/LibreNMS/Snmptrap/Handlers/VeeamTrap.php#L32

Reply:
not sure I understand, Xmatters expects certain values in the json payload "event_state", 
I am translating this from alert_data['state'], if I convert it to enums would not then translate those to a json payload to get an Xmatters readable output?